### ISSUE TITLE
centroid analysis should have option for individual geometry or entire dataset

### ISSUE THREAD

**Issue by ohasselblad** [CartoDB/camshaft#127]
Right now centroid find the centroid of the collected dataset (`ST_Centroid(ST_Collect(the_geom))`, but users typically want to find the centroid of each of the geometries as well (`ST_Centroid(the_geom)`). There should be an option to return one or the other of these, and should probably default to the second one.

cc @rochoa

**Comment by rochoa** [comment]
You can group by cartodb_id, and that gets you a centroid for each of them, right?

**Comment by ohasselblad** [comment]
Yep, that works. To me it feels a little complicated. Could we do an if/then to get different sql templates instead of using st_collect when it's not needed?

**Comment by rochoa** [comment]
I agree, we should have the option. I was proposing a temporal workaround :-).

**Comment by dgaubert** [comment]
>  Could we do an if/then to get different sql templates instead of using st_collect when it's not needed?

So we should use `st_collect` for `points`, otherwise we shouldn't, right?

**Comment by dgaubert** [comment]
> There should be an option to return one or the other of these, and should probably default to the second one.

Opps I didn't read this one. 😓

**Comment by dgaubert** [comment]
> There should be an option to return one or the other of these, and should probably default to the second one

Could we use the current params to figure it out? I mean, if `aggregation` and `aggregation_column` are `null` then don't use `st_collect`.
